robolectric-resources/src/main/java/org/robolectric/res/ResName.java

```diff
@@ -32,7 +32,7 @@ public ResName(@NotNull String fullyQualifiedName) {
       throw new IllegalStateException("\"" + fullyQualifiedName + "\" is not fully qualified");
     }
     packageName = matcher.group(NAMESPACE);
-    type = matcher.group(TYPE);
```

Comment:
Calls to other constructors must be the first line in a constructor, but we first need that Matcher in order to get extract the packageName, type, and name. The next closest thing I can think of is having both constructors call some setup(String, String, String) method that just sets the values of the fields, but I feel like that gets to be a little overkill.